crates/sdk/src/signing.rs

```diff
@@ -1714,6 +1845,30 @@ pub async fn to_ledger_vector(
         tv.output.push("Type : Custom".to_string());
     }
 
+    if tx.memo_sechash() != &namada_core::types::hash::Hash::default() {
+        match tx
+            .get_section(tx.memo_sechash())
+            .unwrap()
+            .extra_data_sec()
+            .unwrap()
+            .code
+        {
+            Commitment::Hash(hash) => {
+                tv.output
+                    .push(format!("Memo Hash : {}", HEXLOWER.encode(&hash.0)));
+                tv.output_expert
+                    .push(format!("Memo Hash : {}", HEXLOWER.encode(&hash.0)));
+            }
+            Commitment::Id(id) => {
+                let memo = String::from_utf8(id).map_err(|err| {
+                    Error::from(EncodingError::Conversion(err.to_string()))
+                })?;
+                tv.output.push(format!("Memo : {}", memo));
+                tv.output_expert.push(format!("Memo : {}", memo));
+            }
```

Comment:
since there is no bound on the size of memos (IIRC), we might run over the length of the columns available to print on in the ledger's screen, no?

Reply:
Good question. Yes you're correct about there being potentially insufficient columns. But this issue is mitigated in the hardware by paging long outputs, i.e. printing only a part of the memo at a time and allowing the user to press "next". To reflect this reality, paging is implemented for the test vectors too. See the function `format_outputs` in `signing.rs` that does this string splitting.